### PR TITLE
Configurable storage path

### LINKED ISSUE #85: Add option to set storage directory
Right now, storage directory is simply "storage/" relative to the CWD when Cardinal is started. This should probably be configurable.

### PR DESCRIPTION
**Pull request by BiohZn**
Suggestion for configurable storage path, fixes #85

### CODE REVIEW

**Comment by BiohZn** [comment]
As a further suggestion to this, the storage directory would be removed from the repo, as it is added into the configured directory on first launch.

**Comment by JohnMaguire** [comment]
Hey thanks for this PR, it looks great!

However, there's one slight thing I'd like fixed first: I've been trying to keep application code / logic out of the CardinalBotFactory, and use it primarily for managing the connection to IRC. I'm thinking the code for creating the storage directory could go one of two places, and I'm not too attached to either one, so perhaps whichever you think makes sense:

1. In the `cardinal.py` file, after all the configuration juggling. Consider it part of the "bootstrapping" process.
2. In `CardinalBot` itself, though I think this could be a little awkward because then we aren't creating the storage directory until AFTER spawning our IRC connection.

One more thing! I think the "Storage path does not exist, creating it..." log line could probably be set to `info()`. Especially since I'd like to see this line if an exception gets thrown while attempting to create the storage path (say, permissions error.)

Thanks a ton for this PR!

**Comment by BiohZn** [comment]
Hey,

And thanks for the feedback :) I think that option 1 sounds better, will fix that at once!

**Comment by JohnMaguire** [comment]
> As a further suggestion to this, the storage directory would be removed from the repo, as it is added into the configured directory on first launch.

I will definitely consider this! One of the really nice parts about this `storage/` configuration option is that it's helping to position Cardinal to be package-able in the normal `setuptools` process, and thus pip installable. What this means is, eventually you won't be working in a Git clone of Cardinal (which gets kind of messy, because you're modifying config, adding plugins, etc. and then you have to hope a `git pull` doesn't conflict with anything locally.)

This probably means `storage/` in the repo will become a little useless anyway -- unless `storage/` becomes sort of multi-tiered (plugins will eventually move to `storage/` I think, and then there might be your upstream [default plugins and what have you] `storage/` directory as well as a user-created one.)

In other words, I think you're right, I just have to give it a little more thought. :+1:

**Comment by JohnMaguire** [comment]
Ah, and one more thing: Be sure to add your name to the `CONTRIBUTORS` file in a separate commit if you would like. :)

**Comment by BiohZn** [comment]
There we go, should be all set now :)

File: cardinal.py [hunk 1]
```diff
@@ -121,12 +123,34 @@
...
+            storage_path = os.path.join(
+                os.path.dirname(os.path.realpath(sys.argv[0])),
+                config['storage']
+            )
+
+        logger.debug("Storage path set to %s" % storage_path)
+
+        if not os.path.exists(storage_path):
+            logger.info("Storage path does not exist, creating it..")
+            for folder in ['database', 'logs']:
```
Comment (BiohZn) [inline cardinal.py]: Cleaner way of doing it, compared to the earlier commit

Comment (JohnMaguire) [inline cardinal.py]: I actually considered this method too, since we might need to create more later. Awesome!

I'll get around to testing this out later today! Thanks again for the pull request! :+1:

**Comment by JohnMaguire** [comment]
Thanks again for this! I made two minor changes before merging, hope that's okay:

1. Calculated the full "realpath" to the storage directory (for logging purposes, mainly).
2. Checked each directory for existence separately -- this way if "storage/" exists, but "storage/database/" doesn't, we'll create it.
